src/java/org/apache/cassandra/streaming/StreamPlan.java

```diff
@@ -63,6 +63,13 @@ public StreamPlan(StreamOperation streamOperation, boolean connectSequentially)
     public StreamPlan(StreamOperation streamOperation, int connectionsPerHost,
                       boolean connectSequentially, UUID pendingRepair, PreviewKind previewKind)
     {
+        this(UUIDGen.getTimeUUID(), streamOperation, connectionsPerHost, connectSequentially, pendingRepair, previewKind);
+    }
+
+    public StreamPlan(UUID planId, StreamOperation streamOperation, int connectionsPerHost,
```

Comment:
the idea is that streaming done by repair will use the RepairJob id (as defined by CASSANDRA-15399), so if a repair exists with the same planId, then this stream can map to the repair without any networking changes